crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -69,4 +70,90 @@ impl TransactionReceipt {
         }
         Some(self.from.create(nonce))
     }
+
+    /// Returns `true` if the transaction was successful.
+    /// A transaction is considered successful if the status code is `1`.
+    fn success(&self) -> bool {
+        match &self.status_code {
```

Comment:
I'm okay with changing the `status_code` from U64 to u64 and use a serde_with hex_opt for this

Reply:
i don't get what does it mean by "use a serde_with hex_opt" in this case. Could you elaborate more?

Reply:
you mean `#[serde(with = "hex")]`?

Reply:
yeah, but we can do this separately

Reply:
given that status is always 0 or 1, I would also be comfortable just deserializing it to a bool. "real" future compatibility would be a non-exhaustive enum imho

Reply:
checked #396 , will leave this nit on that PR

Reply:
ya sorry for stepping on your PR's toes a bit here. I feel like the infallible conversion will serve your needs better? 😅

Reply:
no worries, i like embed approach better. 
one q is I had checked `impl AsRef<ReceiptEnvelope> for TransactionReceipt` meet the goal what this PR had supposed to, should I have to close the PR?

Reply:
we will talk internally, and then close one of them. If you like the embed approach better, we'll likely go in that direction :)

thank you for making the issue and PR regardless!